### PR TITLE
MPI Python Interface

### PR DESCRIPTION
**Pull request by cangumeli**

### CODE REVIEW

**Comment by giangiac** [comment]
Hi Can, I will add an action in which the IQS Python library is built. In this way we can automatically check that part of the building process. I will do so in a separate pull request.

**Comment by fbaru-dev** [comment]
I build the code with and without MPI, run the test prepared by Can, and I do not see any issue. Additional tests will come.

File: src/mpi_env.cpp [hunk 1]
```diff
@@ -163,6 +186,30 @@ Environment::Environment(int& argc, char**& argv) : inited_(false)
...
+  QHIPSTER_MPI_CHECK_RESULT(MPI_Initialized,(&flag))
+  if (!flag) {
+    QHIPSTER_MPI_CHECK_RESULT(MPI_Init,(&argc, &argv))
+    inited_ = true;
+  }
+  CommonInit(flag);
+}
+
+
+// TODO: Copy-pase! Consider refactoring.
```
Comment (fbaru-dev) [inline src/mpi_env.cpp:201]: @cangumeli what do you mean by this comment: consider refactoring?

**Comment by fbaru-dev** [comment]
Merged into development.

**Pull request merged**